rcl/include/rcl/context.h

```diff
@@ -240,6 +240,32 @@ RCL_WARN_UNUSED
 rcl_context_instance_id_t
 rcl_context_get_instance_id(rcl_context_t * context);
 
+/// Returns the context domain id.
+/**
+ * \pre If context is uninitialized, then it is undefined behavior.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | Yes [1]
+ * Uses Atomics       | No
+ * Lock-Free          | No
+ *
+ * <i>[1] Calling the function asynchronously with `rcl_init` or `rcl_shutdown` can result
+ *  in the function sometimes succeeding and sometimes returning `RCL_RET_INVALID_ARGUMENT`.
+ *
+ * \param[in] context object from which the domain id should be retrieved.
+ * \param[out] domain_id output variable where the domain id will be returned.
+ * \return RCL_RET_INVALID_ARGUMENT if `context` is invalid, or
```

Comment:
@ivanpauno nit: it'd be good to specify what `invalid` means (perhaps refer to the function that checks for its validity?)

Reply:
Done in https://github.com/ros2/rcl/pull/718/commits/33fcc78e343dde17b80c407f5b9016ef0ddb9812